src/dynamic_sharing_service/extension/third.party.attribute.extension.builder.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+const Extension = require('./extension');
+const ThirdPartyAttributeExtensionContent = require('./third.party.attribute.extension.content');
+const AttributeDefinition = require('../attribute.definition');
+const Validation = require('../../yoti_common/validation');
+
+const THIRD_PARTY_ATTRIBUTE_EXTENSION_TYPE = 'THIRD_PARTY_ATTRIBUTE';
+
+class ThirdPartyAttributeExtensionBuilder {
+  constructor() {
+    this.definitions = [];
+  }
+
+  withExpiryDate(expiryDate) {
+    Validation.instanceOf(expiryDate, Date, 'expiryDate');
+    this.expiryDate = expiryDate;
+    return this;
+  }
+
+  withDefinition(definition) {
```

Comment:
As discussed, we should pass a string to the method used to create an `AttributeDefinition`

Reply:
Fixed in https://github.com/getyoti/yoti-node-sdk/pull/110/commits/8b8c3f343791dacba873ac9d5eab36cd76575698